client/app/bundles/course/courses/components/buttons/TodoIgnoreButton.tsx

```diff
@@ -0,0 +1,61 @@
+import { defineMessages, injectIntl, WrappedComponentProps } from 'react-intl';
+import { FC } from 'react';
+import { Button } from '@mui/material';
+import { toast } from 'react-toastify';
+import { getCourseId } from 'lib/helpers/url-helpers';
+import { useDispatch } from 'react-redux';
+import { AppDispatch } from 'types/store';
+import { removeTodo } from '../../operations';
+
+interface Props extends WrappedComponentProps {
+  ignoreLink: string;
+  todoType: 'assessments' | 'videos' | 'surveys';
+  todoId: number;
+}
+
+const translations = defineMessages({
+  ignoreSuccess: {
+    id: 'course.todo.ignore.success',
+    defaultMessage: 'Pending task successfully ignored',
+  },
+  ignoreFailure: {
+    id: 'course.todo.ignore.failure',
+    defaultMessage: 'An error occured',
+  },
+});
+
+const TodoIgnoreButton: FC<Props> = (props) => {
+  const { intl, ignoreLink, todoType, todoId } = props;
+  const dispatch = useDispatch<AppDispatch>();
+
+  const onIgnore = (): Promise<void> => {
+    const courseId = getCourseId();
+    if (courseId) {
```

Comment:
Why do we need a check here?

Reply:
courseId can be string | null. Without the check there will be an error
![image](https://user-images.githubusercontent.com/48171632/175205400-cbd0bfe4-013a-4466-a3c8-407d82f9cbae.png)

Reply:
Ah i see, I think I am quite sure we courseId will never be null right?  U can do `getCourseId()!`

Reply:
It is already using getCourseId() haha but the function is declared in a JS file and the type is returned as string | null